### PR TITLE
Adding the functionality to search and replace a line

### PR DESCRIPTION
**Pull request by ovisan**
The search is done with regex expressions and the replace is string
https://github.com/crowbar/crowbar-core/pull/96
reopening for idiomatic chef

### CODE REVIEW

**Comment by tboerger** [comment]
:+1:

**Comment by MaximilianMeister** [comment]
would have made the formatting an extra commit but it doesn't really matter
:+1:

File: chef/cookbooks/utils/providers/line.rb [hunk 1]
```diff
@@ -87,14 +87,40 @@ def add_and_filter(file, add, filter_re)
...
+    nil,
+    "^#{@new_resource.name}$"
+  )
+  @new_resource.updated_by_last_action(true) if updated
+end
+
+action :replace do
+  updated = search_and_replace(
+    @new_resource.file,
+    @new_resource.regexp,
```
Comment (vuntz) [inline chef/cookbooks/utils/providers/line.rb]: There's no `regexp` attribute in https://github.com/ovisan/crowbar-core/blob/searchandreplace/chef/cookbooks/utils/resources/line.rb

Comment (ovisan) [inline chef/cookbooks/utils/providers/line.rb]: @vuntz correct, i don't know what happened, but it's fixed now. Also `:replace` was not defined

**Comment by tboerger** [comment]
Still :+1:

**Comment by rsalevsky** [comment]
+1

**Pull request merged**